plugins/common/proxy/proxy.go

```diff
@@ -4,21 +4,56 @@ import (
 	"fmt"
 	"net/http"
 	"net/url"
+
+	"golang.org/x/net/proxy"
 )
 
 type HTTPProxy struct {
+	UseProxy     bool   `toml:"use_proxy"`
 	HTTPProxyURL string `toml:"http_proxy_url"`
 }
 
 type proxyFunc func(req *http.Request) (*url.URL, error)
 
 func (p *HTTPProxy) Proxy() (proxyFunc, error) {
-	if len(p.HTTPProxyURL) > 0 {
-		address, err := url.Parse(p.HTTPProxyURL)
-		if err != nil {
-			return nil, fmt.Errorf("error parsing proxy url %q: %w", p.HTTPProxyURL, err)
+	if p.UseProxy {
```

Comment:
@reimda, @sspaink and I were chatting and we believe this would break existing users who only have the proxy URL defined. In order to add this in a backwards-compatible way we need to implement a priority as such:

1. system proxy: if the `UseSystemProxy` is enabled, use the system proxy
2. HTTP proxy URL: if the `HTTPProxyURL` is set use this
3. no proxy

We may have users with configs today that only have the URL defined and not the boolean.

Reply:
Sounds good. Should I make `TCPProxy` act the same way for consistency?

Reply:
yes if you have the time :)